driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/crypt/Crypt.java

```diff
@@ -227,8 +232,9 @@ public void close() {
         //noinspection EmptyTryBlock
         try (MongoCrypt ignored = this.mongoCrypt;
              CommandMarker ignored1 = this.commandMarker;
-             MongoClient ignored2 = this.internalClient;
-             KeyManagementService ignored3 = this.keyManagementService
+             MongoClient ignored2 = this.collectionInfoRetrieverClient;
+             MongoClient ignored3 = this.keyVaultClient;
+             KeyManagementService ignored4 = this.keyManagementService
```

Comment:
This is not related to the PR changes, but it's not obvious why the async `KeyManagementService` uses `TlsChannelStreamFactoryFactory` (and needs to be closed), while its sync counterpart does not. Do you happen to know anything about it?

Reply:
I think it's because with async I/O you need a thread pool to handle the I/O completion, and for the library that we're using for async TLS there is no "system" (i.e. global) thread pool that is used (unlike with the JDK's `AsynchrounousSocketChannel`)

Reply:
Thanks!